src/wp-includes/option.php

```diff
@@ -382,6 +382,16 @@ function wp_load_core_site_options( $network_id = null ) {
 
 	$core_options = array( 'site_name', 'siteurl', 'active_sitewide_plugins', '_site_transient_timeout_theme_roots', '_site_transient_theme_roots', 'site_admins', 'can_compress_scripts', 'global_terms_enabled', 'ms_files_rewriting' );
 
+	if ( wp_using_ext_object_cache() ) {
+		$cache_keys = array();
+		foreach ( $core_options as $option ) {
+			$cache_keys[] = "{$network_id}:$option";
+		}
+		wp_cache_get_multiple( $cache_keys, 'site-options' );
```

Comment:
I'm unclear what is happening here, the results are not used.

Reply:
@peterwilsoncc Instead of loading these keys one at a time, it loads them all at once, saving 9 single calls to object cache. This function doesn't return anything, just primes caches.

Reply:
The first time (after wp_cache_flush() or when cache will expire), all cache keys will be empty because there is no cache. So, if the first array item will have an empty value (I have no better solution for this), we need to proceed further to the last wp_cache_set_multiple() function. And because you are broadening the application of this function to include persistent cache, its description needs to be also changed to clarify this confusion. I am also wondering if this can present a regression. Imagine that some key has already expired but the first one is still present, cache will return false for this key and there is no way to tell that false is an absence of value in cache or an actual value. I tried it right now with Redis, and the plugin is returning false in both cases.

Reply:
@OlaIola If I understand your comment. You are worried about regression. I don't see how that could happen. After cache is clear, after `wp_cache_get_multiple` will be called and return nothing. Then in the rest of the request, those values will be populated. One the next request `wp_cache_get_multiple` will work just fine. But it how it works now. this is not a regression.

Reply:
Ok, @spacedmonkey, I tested with Redis. There is no regression I was able to find, we are not actually using this data straight away, so, it should be fine and there is some speed improvement in retrieving data after wp_cache_get_multiple() as well if persistent cache exists. So, the only thing is that the comment to the function (and possibly to this part of the code) should be updated to actual state and to remove uncertainty of what is actually happening.

Reply:
@OlaIola Good catch on the PHPDoc block. I have updated the doc block now. I know the naming of network and site is confusing. Network is the current name of networks. Network were called sites at one point that changed many releases ago. Nice benefit of this change is to fix this documentations.

Reply:
> This function doesn't return anything, just primes caches.

Thanks, I understand now.

Reply:
I think we might need something to check that all values are in the cache.

If `site_name` and `siteurl` are cached but the remaining values have fallen out of it, then it seems that data will be missing and it would be better to query the DB for those options and add them to the cache.

Let me know if I am missing something.

Reply:
@peterwilsoncc Sadly, that is not possible. As you know we don't have a `$found` parameter to `wp_cache_get_multiple`. See #4258. Some of the values of valid false. So it hard to know if we got a value here at all. 

What you are worrying about here is super edge case. These values are highly likely to be in the cache. If they are not, then when they are requested in get_network_option, they will be found missing in the cache and will be loaded from the database. 

The happy path of this code of the values being in the cache and this primes the caches in one go, is extremely likely. Any other changes are just adding extra bloat to this bootstrap function that will not really add value but will add complexity and compute.

Reply:
When I first commented, I also thought that we can check if `site_name` (or other option which should not be empty) is in the cache and if not, not to return but to go further to collect all data from the DB to set the cache, but this way all cached values will be settled no matter if they existed in the cache or not. Possibly it can be an option, but the question is what is best for performance.

Reply:
We could maybe use the $found param on `wp_cache_get` like this. 
```
$not_found_keys = array();
foreach ( $core_options as $option ) {
    $found = null;
    wp_cache_get( "{$network_id}:{$option}", 'site-options', $found );
    if( false === $found ) {
       $not_found_keys[] = $option;
    }
}

if( ! $not_found_keys ) {
    return;
}

$core_options = $not_found_keys;
```

This would only prime not found options. 

This feel like overkill, but it would be possible.

Reply:
@spacedmonkey yeah, it is a bit much. If we have no side effects and performance is better for persistent cache, maybe we can apply your patch without too much hustle? :)

Reply:
My reading of the code is that `wp_load_core_site_options()` is only run during bootstrapping, ie getting data from the cache is only applicable to sites with a persistent cache.

As the meta data and options tables store data in string form, any false values will be returned as `0` rather than `false`.

If the returned value from `wp_cache_get_multiple()` is `=== false` I think it's certain that the data is not stored in the persistent cache.

This is the test I used:

```php
public function test_register_setting_with_genuine_false_value() {
	register_setting(
		'test_group',
		'pwcc',
		array(
			'type' => 'boolean',
			'sanitize_callback' => function( $value ) {
				return (bool) $value;
			},
			'default' => false,
		)
	);

	$this->assertFalse( get_option( 'pwcc' ) );
	update_option( 'pwcc', true, false );
	$this->assertTrue( get_option( 'pwcc' ) );
	$found = false;
	$actual = wp_cache_get( 'pwcc', 'options', false, $found );
	global $wpdb;
	$row = $wpdb->get_row( $wpdb->prepare( "SELECT option_value FROM $wpdb->options WHERE option_name = %s LIMIT 1", 'pwcc' ) );
	var_dump( $actual, $found, $row );
	/*
	bool(true)
	bool(true)
	object(stdClass)#123 (1) { ["option_value"]=> string(1) "1" }
	*/

	wp_cache_flush();
	get_option( 'pwcc' );
	$found = false;
	$actual = wp_cache_get( 'pwcc', 'options', false, $found );
	global $wpdb;
	$row = $wpdb->get_row( $wpdb->prepare( "SELECT option_value FROM $wpdb->options WHERE option_name = %s LIMIT 1", 'pwcc' ) );
	var_dump( $actual, $found, $row );
	/*
	string(1) "1"
	bool(true)
	object(stdClass)#123 (1) { ["option_value"]=> string(1) "1" }
	*/
}
```

Reply:
@peterwilsoncc Not sure what what you are asking for here? I am saying that I think that checking the cache and loading via database is not a good idea. It only happens the first page load after a cache flush and make the code harder to understand and more brittal. 

With Beta 1 tomorrow, can we commit this change and iterate over it?

Reply:
I agree with @OlaIola. If it is not a blocker, let's commit it now and address it in a future iteration.

Reply:
Also agreed. If this improves performance for what _is_ in the persistent cache, and makes no difference to performamce for what _isn't_ in the cache and introduces no bugs, then there's a clear benefit, even if we'd like to improve on that later.